File: android/src/main/java/io/mosip/tuvali/wallet/characteristics/DisconnectCharacteristic.kt
package io.mosip.tuvali.wallet.characteristics

import io.mosip.tuvali.transfer.CRCValidator
import io.mosip.tuvali.verifier.GattService
import java.util.*

class DisconnectCharacteristic(val data: ByteArray) {
  val status = data[0].toInt()

  companion object{
    val uuid: UUID = GattService.DISCONNECT_CHAR_UUID
  }

  init{
    CRCValidator.validateDataFromVerifier(data, uuid.toString())
  }
}

File: ios/Wallet/characteristics/DisconnectCharacteristic.swift

import Foundation
import CoreBluetooth


class DisconnectCharacteristic {
  var data: Data
  static let uuid = CharacteristicIds.DISCONNECT_CHAR_UUID
  var status:  Int

    init(data: Data){
        self.data =  data
        status = Int(data[0])
         CRCValidator.verify(data: data, characteristic: DisconnectCharacteristic.uuid.rawValue)
    }
}
